test/unit/classes/decorators.spec.ts

```diff
@@ -126,3 +126,86 @@ test("Exported class decorator", () => {
         .setReturnExport("Foo", "bar")
         .expectToMatchJsResult();
 });
+
+test.each([
+    ["method() {}"],
+    ["property;"],
+    ["propertyWithInitializer = () => {};"],
+    ["['evaluated property'];"],
+    ["get getter() { return 5 }"],
+    ["set setter(value) {}"],
+    ["static method() {}"],
+    ["static property;"],
+    ["static propertyWithInitializer = () => {}"],
+    ["static get getter() { return 5 }"],
+    ["static set setter(value) {}"],
+    ["static ['evaluated property'];"],
+])("Decorate class member (%p)", memberStatement => {
+    util.testFunction`
+        let decoratorTarget: any | undefined;
+        let decoratorTargetKey: string | undefined;
+
+        const decorator = (target, key) => {
+            decoratorTarget = target;
+            decoratorTargetKey = key;
+        };
+
+        class Foo {
+            @decorator
+            ${memberStatement}
+        }
+
+        const targetKind = decoratorTarget === Foo.prototype ? "prototype" : "class";
+        return { targetKind, decoratorTargetKey };
```

Comment:
I'd merge this and the next test into something like that:
```ts
		let decoratorParameters: any;
		const decorator = (target, key, index) => {
			const targetKind = target === Foo ? "Foo" : target === Foo.prototype ? "Foo.prototype" : "unknown";
			decoratorParameters = { targetKind, key, index };
        };

		class Foo {
			${classMember}
		}

		return decoratorParameters;
```

Reply:
Good idea, did something like this with 9f88c3f